bazel_tools/nixpkgs-hermetic-cc-toolchain.patch

```diff
@@ -0,0 +1,1221 @@
++    expand_location = True,
++    nix_file = "//tests:location_expansion.nix",
++    nix_file_deps = [
++        "//:nixpkgs.json",
++        "//:nixpkgs.nix",
++        "@io_tweag_rules_nixpkgs//tests:relative_imports/nixpkgs.nix",
++    ],
++    nixopts = [
++        "--arg",
++        "attrs",
++        "{ nixpkgs_json = $(location //:nixpkgs.json); nixpkgs_nix = $(location //:nixpkgs.nix); }",
++        "--arg",
++        "relative_imports",
++        "$(location @io_tweag_rules_nixpkgs//tests:relative_imports/nixpkgs.nix)",
++    ],
++    repository = "@remote_nixpkgs",
++)
++
+ load("@bazel_tools//tools/build_defs/repo:http.bzl", "http_archive")
+ 
+ http_archive(
+@@ -179,12 +200,11 @@ http_archive(
+ 
+ load(
+     "//nixpkgs:toolchains/go.bzl",
+-    "nixpkgs_go_configure"
++    "nixpkgs_go_configure",
+ )
+ 
+ nixpkgs_go_configure(repository = "@nixpkgs")
+ 
+-load("@io_bazel_rules_go//go:deps.bzl", "go_rules_dependencies", "go_register_toolchains")
++load("@io_bazel_rules_go//go:deps.bzl", "go_register_toolchains", "go_rules_dependencies")
+ 
+ go_rules_dependencies()
+-
+diff --git a/nixpkgs/constraints/BUILD.bazel b/nixpkgs/constraints/BUILD.bazel
+index 6662530..94678bd 100644
+--- a/nixpkgs/constraints/BUILD.bazel
++++ b/nixpkgs/constraints/BUILD.bazel
+@@ -3,7 +3,7 @@ package(default_visibility = ["//visibility:public"])
+ constraint_setting(name = "nix")
+ 
+ constraint_value(
+-    name =  "support_nix",
++    name = "support_nix",
+     constraint_setting = ":nix",
+ )
+ 
+diff --git a/nixpkgs/nixpkgs.bzl b/nixpkgs/nixpkgs.bzl
+index fb87da4..6bf2d90 100644
+--- a/nixpkgs/nixpkgs.bzl
++++ b/nixpkgs/nixpkgs.bzl
+@@ -1,7 +1,15 @@
+ """Rules for importing Nixpkgs packages."""
+ 
++load("@bazel_skylib//lib:paths.bzl", "paths")
++load("@bazel_skylib//lib:sets.bzl", "sets")
+ load("@bazel_tools//tools/cpp:cc_configure.bzl", "cc_autoconf_impl")
+-load("@bazel_tools//tools/cpp:lib_cc_configure.bzl", "get_cpu_value")
++load(
++    "@bazel_tools//tools/cpp:lib_cc_configure.bzl",
++    "get_cpu_value",
++    "get_starlark_list",
++    "write_builtin_include_directory_paths",
++)
++load("@bazel_tools//tools/build_defs/repo:utils.bzl", "maybe")
+ 
+ def _nixpkgs_git_repository_impl(repository_ctx):
+     repository_ctx.file("BUILD")
+@@ -70,6 +78,54 @@ nixpkgs_local_repository = repository_rule(
+ def _is_supported_platform(repository_ctx):
+     return repository_ctx.which("nix-build") != None
+ 
++def _expand_location(repository_ctx, string, labels, attr = None):
++    """Expand `$(location label)` to a path.
++
++    Attrs:
++      repository_ctx: The repository rule context.
++      string: string, Replace instances of `$(location )` in this string.
++      labels: dict from label to path: Known label to path mappings.
++      attr: string, The rule attribute to use for error reporting.
++
++    Returns:
++      The string with all instances of `$(location )` replaced by paths.
++    """
++    num = string.count("$(location ")
```

Comment:
This looks like it is inconsistent with how Bazel handles this kind of stuff where you have to escape the `$` if you want it to not have special meaning. Probably not a big deal.

Reply:
I don't follow. We do want `$(location )` to have special meaning, just that here we define that special meaning with `_expand_location`. Note `$(location )` expension is not automatic, a rule has to invoke [`ctx.expand_location`](https://docs.bazel.build/versions/master/skylark/lib/ctx.html#expand_location) explicitly. However, here we are within a repository rule, so `ctx.expand_location` is not available to us.

Reply:
What I mean is that only `$(location …)` has special meaning. In other contexts in Bazel where `$(location …)` has special meaning, everything starting with `$` has special meaning and you have to escape something like `$abc` whereas here you do not need the escaping.

Reply:
Oh, I see, yes that would be more consistent. location expansion is not enabled by default, so it's not a big deal here. But, yes, this should be improved upstream.